src/test/java/teammates/storage/api/FeedbackResponseMonitorDbTest.java

```diff
@@ -0,0 +1,58 @@
+package teammates.storage.api;
+
+import java.util.List;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.FeedbackResponseRecordAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.test.BaseComponentTestCase;
+
+/**
+ * SUT: {@link FeedbackResponseMonitorDb}.
+ */
+public class FeedbackResponseMonitorDbTest extends BaseComponentTestCase {
+    FeedbackResponseMonitorDb db = new FeedbackResponseMonitorDb();
+    long[] countTestData = new long[] { 105, 105, 105, 105, 105 };
+
+    private void populateTestData() throws EntityAlreadyExistsException, InvalidParametersException {
+        long currentTime = System.currentTimeMillis();
+        for (int i = 0; i < countTestData.length; i++) {
+            long timestamp = currentTime - 120 * i;
+            FeedbackResponseRecordAttributes attributes = new FeedbackResponseRecordAttributes(countTestData[i], timestamp);
+            db.createEntity(attributes);
+        }
+    }
+
+    @Test
+    public void testResponseRecordsInterval() throws EntityAlreadyExistsException, InvalidParametersException {
+        long interval = 120;
+        db.purgeResponseRecords();
+        populateTestData();
+        //get all records with interval 120 milliseconds
+        List<FeedbackResponseRecordAttributes> results =
+                db.getResponseRecords(System.currentTimeMillis(), interval);
+        long currentTime = -1;
+        for (FeedbackResponseRecordAttributes attributes : results) {
+            if (currentTime != -1) {
+                //validate interval
+                assertEquals(attributes.getTimestamp() - currentTime, 120);
+            }
+            currentTime = attributes.getTimestamp();
+        }
+    }
+
+    @Test
+    public void testResponseRecordsDuration() throws EntityAlreadyExistsException, InvalidParametersException {
```

Comment:
In general, this test doesn't look good:
What if someday, the test data contains an entry that is exactly 400 (milli?)seconds before the current time (from above) or with some lag, +-10 or so, i.e. a record that sits at the border of the duration, and then just nice the db query takes slightly longer in some runs (CI instance might have higher network latency **sometimes**) and the test fails **sometimes**. This means that the test might enter an unstable state.

Reply:
thats not possible, because the current time is taken once from the system, and other test data is generated based on that (with custom interval) so the duration between current time and oldest time will always be accurate

Reply:
I have made the ```currentTime``` a final class member, so test data should be stable now

Reply:
Ok this looks good. See the other comment above